core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/CarbonUIUtil.java

```diff
@@ -510,4 +543,71 @@ public static String getDefaultManagementUIPath() {
     private static Object getDefaultHomePageProductParam() {
         return getProductParam(CarbonConstants.PRODUCT_XML_WSO2CARBON + CarbonConstants.DEFAULT_HOME_PAGE);
     }
+
+
+    /**
+     * Returns absolute URL of admin console webapp for given relative path
+     * if IS_RESOLVE_ABSOLUTE_URLS_ENABLED config is enabled.
+     *
+     * @param context       Webapp context root of the Carbon webapp.
+     * @param relativePath  Relative path of the Carbon webapp
+     * @param request       Request that used to redirect.
+     * @return absolute URL of admin console webapp for given relative path.
+     */
+    public static String resolveAdminConsoleBaseURL(String context, String relativePath, HttpServletRequest request) {
+
+        if (isResolveAbsoluteURLsEnabled()) {
+
+            // Removing any tailing "/" in the context.
+            context = getAdminConsoleBaseURL(context);
+            if (context.endsWith("/")) {
+                context = context.substring(0, context.length() - 1);
+            }
+
+            // Remove any tailing "/carbon" in context to build base URL.
+            if (context.endsWith("/carbon")) {
+                context = context.substring(0, context.length() - 7);
+            }
+
+            // Build relative path starting from root context.
+            List<String> splitPathList = new ArrayList<>(Arrays.asList(
+                    request.getContextPath().concat(request.getServletPath()).split("/")));
+            splitPathList.remove(0);
+            // If the request is a base URL, add the carbon as the root context.
+            if (splitPathList.isEmpty()) {
+                splitPathList.add("carbon");
+            }
+
+            // Replace ".." with the node of path directory.
+            int index = 0;
+            while (relativePath.contains("..")) {
+                relativePath = relativePath.replaceFirst("..", splitPathList.get(index));
+                index++;
+            }
+
+            // Add "/", if relative path is not starting with.
+            if (relativePath.charAt(0) != '/') {
+                relativePath = "/" + relativePath;
+            }
+        }
+
+        return context + relativePath;
+    }
+
+    /**
+     * Returns whether resolving absolute URL config is enabled or not.
+     *
+     * @return Resolving absolute URL config is enabled.
+     */
+    public static boolean isResolveAbsoluteURLsEnabled() {
+
+        String  isResolveAbsoluteURLsEnabled = CarbonUIServiceComponent.getServerConfiguration()
+                .getFirstProperty(CarbonConstants.IS_RESOLVE_ABSOLUTE_URLS_ENABLED);
+
+        if (isResolveAbsoluteURLsEnabled == null) {
```

Comment:
should this be a null check or blank string check?

Reply:
When the isResolveAbsoluteURLsEnabled is blank, it will be handled by the line 611 with Boolean.parseBoolean() and will return the false.